spec/remote_fetching_spec.rb

```diff
@@ -102,6 +110,16 @@
     end
   end
 
+  context 'when the server requires authentication' do
```

Comment:
I feel this scenario is fairly detailed—authorization is just one use-case for using headers. Wouldn't it be better to generically test whether headers can be passed along with the request?

Reply:
I rephrased the test. Do you think that it's enough?

To test that the headers were passed, I would have to stub/mock `Faraday` instance before calling `FormatParser.parse_http`, but it wouldn't work because `Faraday` is required only during execution time.

Do you have other suggestion on how to test it?

Reply:
I think naming "authorization" to "test_header" or something similar is already a good step towards abstracting this.

Reply:
I think I found a way for that -> 1754f33

Could you take a look pls?